### PR TITLE
Solution for issue 336

### PR DESCRIPTION
**Pull request by hlocon1309**
Solution for issue 336 has the following steps:
1. Locate command line arguments related to app options (map -> options)
2. Locate app specific options (vector -> appOptions)
3. Check if command line options match with appOptions
4. If don't match, exit with an error

### CODE REVIEW

**Review by hlocon1309 — COMMENTED** [review]
Final version October 2023

**Comment by kfattila** [comment]
Hi Bill, could you please test the code. and maybe make some suggestion in the error messages, etc.

**Comment by wsnoble** [comment]
I can't get it to compile.  Error message is below.  This is using the same environment that I've used previously, including gcc/10.2.0. Do I need to use a different gcc?

```
In file included from /net/gs/vol3/software/modules-sw/gcc/8.2.0/Linux/CentOS7/x86_64/include/c++/8.2.0/bits/regex_executor.h:258,
                 from /net/gs/vol3/software/modules-sw/gcc/8.2.0/Linux/CentOS7/x86_64/include/c++/8.2.0/regex:63,
                 from /net/noble/vol1/home/noble/proj/crux/crux-toolkit/build-pr661/ext/build/src/percolator/src/Scores.cpp:32:
/net/gs/vol3/software/modules-sw/gcc/8.2.0/Linux/CentOS7/x86_64/include/c++/8.2.0/bits/regex_executor.tcc: In member function ‘bool std::__detail::_Backref_matcher<_BiIter, std::__cxx11::regex_traits<_Ch_type> >::_M_apply(_BiIter, _BiIter, _BiIter, _BiIter)’:
/net/gs/vol3/software/modules-sw/gcc/8.2.0/Linux/CentOS7/x86_64/include/c++/8.2.0/bits/regex_executor.tcc:369:16: error: ‘__equal4’ is not a member of ‘std’
    return std::__equal4(__expected_begin, __expected_end,
                ^~~~~~~~
/net/gs/vol3/software/modules-sw/gcc/8.2.0/Linux/CentOS7/x86_64/include/c++/8.2.0/bits/regex_executor.tcc:369:16: note: suggested alternative: ‘__equal’
    return std::__equal4(__expected_begin, __expected_end,
                ^~~~~~~~
                __equal
/net/gs/vol3/software/modules-sw/gcc/8.2.0/Linux/CentOS7/x86_64/include/c++/8.2.0/bits/regex_executor.tcc:373:14: error: ‘__equal4’ is not a member of ‘std’
  return std::__equal4(__expected_begin, __expected_end,
              ^~~~~~~~
/net/gs/vol3/software/modules-sw/gcc/8.2.0/Linux/CentOS7/x86_64/include/c++/8.2.0/bits/regex_executor.tcc:373:14: note: suggested alternative: ‘__equal’
  return std::__equal4(__expected_begin, __expected_end,
              ^~~~~~~~
              __equal
make[5]: *** [src/CMakeFiles/perclibrary.dir/Scores.cpp.o] Error 1
make[4]: *** [src/CMakeFiles/perclibrary.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [ext/build/src/percolator-stamp/percolator-build] Error 2
make[1]: *** [ext/CMakeFiles/percolator.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by wsnoble** [comment]
Minor point: can we remove the extraneous commas in this error message?

```
FATAL: Options found that don't match tide-search application:
	-> peptide_mass_tolerance used in: , comet , pipeline
```

It should say something like

```
FATAL: Options found that don't match tide-search application:
	-> peptide_mass_tolerance is used in comet, pipeline
```

File: doc/user/release-notes.html [hunk 1]
```diff
@@ -54,6 +54,8 @@ <h2>Release notes for Crux</h2>
 
 <h4>Minor changes</h4>
 <ul>
+  <li>10 Oct 2023: Fixed issue 336 - When a parameter introduced by user doesn't match, app halts with an error.</li>
```
Comment (wsnoble) [inline doc/user/release-notes.html:57]: Delete "Fixed issue 336 -"

**Review by wsnoble — APPROVED** [review]
Looks good!

**Comment by hlocon1309** [comment]
Changes made for extraneous commas and release-notes.

**Comment by kfattila** [comment]
@wsnoble and @CharlesEGrant, This PR is still not merged? There is a conflict in the CreateDocs.cpp. I am not familiar with that code. Charles, could you please check and try to merge that conflict in CreateDocs.cpp if you are familiar with Jaxs?

**Pull request merged**